src/Parallel/TypeTraits.hpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines type traits related to Charm++ types
+
+#pragma once
+
+#include <charm++.h>
+
+namespace Parallel {
+
+/// \ingroup Parallel
```

Comment:
I would say yes, let's get those out of there and stick them here. If you want to do it in this PR (probably easiest) or make an issue to remind us is up to you.